namer/dnssrv/src/main/scala/io/buoyant/namer/dnssrv/DnsSrvNamer.scala

```diff
@@ -0,0 +1,72 @@
+package io.buoyant.namer.dnssrv
+
+import java.net.InetSocketAddress
+
+import com.twitter.finagle._
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.logging.Logger
+import com.twitter.util.Activity.State
+import com.twitter.util._
+import org.xbill.DNS
+
+class DnsSrvNamer(prefix: Path, resolver: DNS.Resolver, timer: Timer, refreshInterval: Duration, stats: StatsReceiver)
+  extends Namer {
+
+  override def lookup(path: Path): Activity[NameTree[Name]] = memoizedLookup(path)
+
+  private val log = Logger.get("dnssrv")
+  private val memoizedLookup: (Path) => Activity[NameTree[Name]] = Memoize {
+    case path@Path.Utf8(address, _) =>
```

Comment:
ok. I actually wrote it that way first, but thought it looked a bit nicer like this because it's one fewer level of indentation (though I guess `Utf8.unapply()` might need to do more work.